net/net.go

```diff
@@ -576,27 +592,31 @@ func (n *net) AddReplicator(ctx context.Context, id thread.ID, paddr ma.Multiadd
 		return
 	}
 
-	// Update peerstore address
-	dialable, err := getDialable(paddr)
-	if err == nil {
-		n.host.Peerstore().AddAddr(pid, dialable, pstore.PermanentAddrTTL)
-	} else {
-		log.Warnf("peer %s address requires a DHT lookup", pid)
-	}
+	// Check if we're dialing ourselves (regardless of addr)
+	if pid.Pretty() != n.host.ID().Pretty() {
+		// If not, update peerstore address
+		var dialable ma.Multiaddr
+		dialable, err = getDialable(paddr)
+		if err == nil {
+			n.host.Peerstore().AddAddr(pid, dialable, pstore.PermanentAddrTTL)
+		} else {
+			log.Warnf("peer %s address requires a DHT lookup", pid)
+		}
 
-	// Send all logs to the new replicator
-	for _, l := range info.Logs {
-		if err = n.server.pushLog(ctx, info.ID, l, pid, info.Key.Service(), nil); err != nil {
-			for _, lg := range managedLogs {
-				// Rollback this log only and then bail
-				if lg.ID.Pretty() == l.ID.Pretty() {
-					if err := n.store.SetAddrs(info.ID, lg.ID, lg.Addrs, pstore.PermanentAddrTTL); err != nil {
-						log.Errorf("error rolling back log address change: %s", err)
+		// Send all logs to the new replicator
```

Comment:
Only push to the replicator if that replicator isn't _us_.

Reply:
It doesn't _really_ make sense to add a replicator that is "us"... but in a multi-tenant situation, the peer calling `AddReplicator` shouldn't know or care if a given peer (that they'd like to add as replicator) is on the same machine.